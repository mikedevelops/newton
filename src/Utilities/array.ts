import { randomInRange } from './number';

/**
 * Get a set of random unique indices from an array
 * @param collection
 * @param amount
 */
export const getUniqueRandomIndices = (collection: any[], amount: number) => {
    const indices = [];
    const max = amount >= collection.length ? collection.length : amount;

    for (let index = 0; index < max; index++) {
        const getRandomIndex = randomInRange.bind(this, 0, collection.length - 1);
        let randomIndex;

        do {
            randomIndex = getRandomIndex();
        } while (indices.indexOf(randomIndex) !== -1);

        indices.push(randomIndex);
    }

    return indices;
};
