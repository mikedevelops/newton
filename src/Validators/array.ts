/**
 * Validate an array has at least 1 item
 * @param array
 */
export const validateAtLeastOneItem = (array: any[]): boolean => {
    if (array.length === 0) {
        throw new Error('Array must contain at least 1 item');
    }

    return true;
};
