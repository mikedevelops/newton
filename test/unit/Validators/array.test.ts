import { validateAtLeastOneItem } from '../../../src/Validators/array';

describe('Array Validators', () => {
    describe('validateAtLeastOneItem', () => {
        test('should throw on an empty array', () => {
            expect(validateAtLeastOneItem.bind(null, [])).toThrowErrorMatchingSnapshot();
        });

        test('should return true for an array with at least one item', () => {
            expect(validateAtLeastOneItem(['foo'])).toBe(true);
        });
    });
});
