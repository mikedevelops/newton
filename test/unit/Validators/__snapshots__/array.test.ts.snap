// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array Validators validateAtLeastOneItem should throw on an empty array 1`] = `"Array must contain at least 1 item"`;
